enginetest/queries/index_queries.go

```diff
@@ -4110,6 +4110,51 @@ var IndexQueries = []ScriptTest{
 			},
 		},
 	},
+	{
+		// MySQL allows creating multiple indexes over the same set of columns. This isn't generally
+		// useful, but some customers need this support. For example, generated migration code from
+		// Django can create cases that require this: https://github.com/dolthub/dolt/issues/8254
+		Name: "multiple indexes over same set of columns",
+		SetUpScript: []string{
+			"CREATE TABLE `t0` (`id` char(32) NOT NULL PRIMARY KEY, `col1` varchar(255) NOT NULL, `col2` varchar(255) NOT NULL);",
+		},
+		Assertions: []ScriptTestAssertion{
+			// Add two indexes over the same column set to t0
+			{
+				Query:    "ALTER TABLE t0 ADD CONSTRAINT unique_1 UNIQUE(col1, col2);",
+				Expected: []sql.Row{{types.NewOkResult(0)}},
+			},
+			{
+				Query:    "ALTER TABLE t0 ADD CONSTRAINT unique_2 UNIQUE(col1, col2);",
```

Comment:
MySQL throws a deprecated warning. Maybe we should too?
```
Warning (code 1831): Duplicate index 'i_3' defined on the table 'test_db.t'. This is deprecated and will be disallowed in a future release.
```

Reply:
Added in 0fd0947fb3d859293d2cb67cf78521dc22b08647